src/components/instantlaunches/InstantLaunchButtonWrapper.js

```diff
@@ -75,6 +80,12 @@ function InstantLaunchButtonWrapper(props) {
                 } else {
                     setOpen(false);
                 }
+                if (autolaunch) {
+                    // go to the analysis landing, not keeping this page in browser history
+                    router.replace(
+                        `/${NavigationConstants.ANALYSES}/${analysis?.id}`
+                    );
+                }
```

Comment:
If this is going to navigate away from this page immediately in the launch success callback, then I see no need to call that `setIlUrl` above, which is intended to open a new window in a `useEffect` after that `setIlUrl` state changes in this success callback. I would think the page would navigate away before that `useEffect` could open that URL.

So I would move this up and make it the first check in the if-else block:
```javascript
                if (autolaunch) {
                    // go to the analysis landing, not keeping this page in browser history
                    router.replace(
                        `/${NavigationConstants.ANALYSES}/${analysis?.id}`
                    );
                } else if (analysis.interactive_urls?.length > 0) {
```

On the other hand, if the VICE URL should still open in a new window on successful launch, then maybe this `router.replace` call should be moved into that `useEffect` after the `ilUrl` is opened in a new window.

Reply:
Ah, yeah, both things should still happen, good point. This redirect is there because, unlike any other place you'd open an instant launch (dashboard, sidebar, instant launch listing, data listing), there's not really any content on the autolaunch page, so without this it opens the VICE URL in a new tab but just shows a blank DE page. I figured taking users to the page where they could stop the analysis or such would be the most useful place to send them, especially since they may have come from a link outside the DE/not already have a DE tab open for notifications, termination, etc.

Reply:
I moved the `router.replace` into the `useEffect`, hopefully correctly (worked when I tested it, at least)